spring-cloud-gcp-bigquery/src/main/java/com/google/cloud/spring/bigquery/core/BigQueryTemplate.java

```diff
@@ -285,6 +285,9 @@ public ListenableFuture<WriteApiResponse> writeJsonStream(
                 writeApiFutureResponse.setException(e);
                 // Restore interrupted state in case of an InterruptedException
                 Thread.currentThread().interrupt();
+              } catch (Throwable t) {
```

Comment:
@elefeint Catching a wider exception is a good idea here and thanks for making the changes. However shall we catch `Exception` instead? Catching `Throwable` results in this Sonar issue: https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=1285&id=GoogleCloudPlatform_spring-cloud-gcp&open=AYPCUFuggImngqXU1xdj

Reply:
Sonar is right, in general, however the user experience will be significantly worse if we follow this recommendation -- for example, if an unexpected Exception (or even `Throwable` such as linkage errors) arises during the separate thread execution, the error will just disappear without leaving a trace.

Reply:
That causes 2 issues:
1) The error disappears without any user-visible evidence.
2) `CompletableFuture` will be left hanging forever.

Reply:
Makes sense @elefeint